source/fundamentals/crud/read-operations/count.txt

```diff
@@ -0,0 +1,256 @@
+.. _csharp-count-documents:
+
+===============
+Count Documents
+===============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to get an :ref:`accurate
+<csharp-accurate-count>` and :ref:`estimated <csharp-estimated-count>` count of
+the number of documents in your collection.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the following documents in a collection called
+``grades``:
+
+.. code-block:: json
+ 
+   { "_id": 1, "name": "Spongebob", "finalGrade": 87.5}
+   { "_id": 2, "name": "Patrick", "finalGrade": 12.3}
+   { "_id": 3, "name": "Squidward", "finalGrade": 99.0}
+   { "_id": 4, "name": "Sandy", "finalGrade": 85.5}
+   { "_id": 5, "name": "Plankton", "finalGrade": 72.3}
+   { "_id": 6, "name": "Karen", "finalGrade": 88.8}
+
+The following ``Student`` class is a Plain Old CLR/Class Object, or
+:ref:`poco <csharp-poco>`,  that models the documents in this
+collection.
+
+.. literalinclude:: /includes/code-examples/count-documents/Student.cs
+   :start-after: start-grades-struct
+   :end-before: end-grades-struct
+   :language: csharp
+   :dedent:
+
+.. note::
+
+   The documents in the ``grades`` collection use the camel-case naming
+   convention. The examples in this guide use a ``ConventionPack``
+   to deserialize the fields in the collection into Pascal case and map them to
+   the properties in the ``Student`` class. 
+
+   To learn more about custom serialization, see
+   :ref:`csharp-custom-serialization`. 
+   
+.. _csharp-accurate-count:
+
+Accurate Count
+--------------
+
+To count the number of documents that match your :ref:`query filter <csharp-specify-query>`, use the
+``CountDocuments()`` method. If you pass an empty query filter, this method
+returns the total number of documents in the collection.
+
+Example 
+~~~~~~~
+
+The following example counts the number of documents where the
+``finalGrade`` is less than ``80``:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
```

Comment:
S: let's create a code file and use includes here instead of copy pasting code. We can discuss synchronously